testing-common/src/main/groovy/io/opentelemetry/instrumentation/test/base/HttpClientTest.groovy

```diff
@@ -814,270 +377,63 @@ abstract class HttpClientTest<REQUEST> extends InstrumentationSpecification {
    * propagate trace context.
    */
   def "high concurrency test"() {
-    setup:
     assumeTrue(testCausality())
-    int count = 50
-    def method = 'GET'
-    def url = resolveAddress("/success")
-
-    def latch = new CountDownLatch(1)
-
-    def pool = Executors.newFixedThreadPool(4)
-
-    when:
-    count.times { index ->
-      def job = {
-        latch.await()
-        runWithSpan("Parent span " + index) {
-          Span.current().setAttribute("test.request.id", index)
-          doRequest(method, url, ["test-request-id": index.toString()])
-        }
-      }
-      pool.submit(job)
-    }
-    latch.countDown()
-
-    then:
-    assertTraces(count) {
-      count.times { idx ->
-        trace(idx, 3) {
-          def rootSpan = it.span(0)
-          //Traces can be in arbitrary order, let us find out the request id of the current one
-          def requestId = Integer.parseInt(rootSpan.name.substring("Parent span ".length()))
-
-          span(0) {
-            name "Parent span " + requestId
-            kind SpanKind.INTERNAL
-            hasNoParent()
-            attributes {
-              "test.request.id" requestId
-            }
-          }
-          clientSpan(it, 1, span(0), method, url)
-          serverSpan(it, 2, span(1)) {
-            it."test.request.id" requestId
-          }
-        }
-      }
-    }
+    expect:
+    junitTest.highConcurrency()
   }
 
   def "high concurrency test with callback"() {
-    setup:
     assumeTrue(testCausality())
```

Comment:
Is it necessary to duplicate assumptions? The JUnit delegate test already calls them.

Reply:
Unfortunately, the jupiter assumption exception is not recognized by junit4 / spock :(